akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -86,22 +85,25 @@ import akka.util.OptionVal
     if (Behavior.isUnhandled(b)) unhandled(msg)
     else {
       b match {
-        case s: StoppedBehavior[T] ⇒
-          // use StoppedBehavior with previous behavior or an explicitly given `postStop` behavior
-          // until Terminate is received, i.e until postStop is invoked, and there PostStop
-          // will be signaled to the previous/postStop behavior
-          s.postStop match {
-            case OptionVal.None ⇒
-              // use previous as the postStop behavior
-              behavior = new Behavior.StoppedBehavior(OptionVal.Some(behavior))
-            case OptionVal.Some(postStop) ⇒
-              // use the given postStop behavior, but canonicalize it
-              behavior = new Behavior.StoppedBehavior(OptionVal.Some(Behavior.canonicalize(postStop, behavior, ctx)))
-          }
-          context.stop(self)
         case f: FailedBehavior ⇒
           // For the parent untyped supervisor to pick up the exception
           throw new TypedActorFailedException(f.cause)
+        case s if !Behavior.isAlive(s) ⇒
+          val lastBehavior = behavior
+          behavior = new ExtensibleBehavior[T] {
+            override def receive(ctx: TypedActorContext[T], msg: T): Behavior[T] =
+              throw new IllegalStateException("Stopping, should never receieve a message")
+            override def receiveSignal(ctx: TypedActorContext[T], msg: Signal): Behavior[T] = {
+              // FIXME the old logic checked if there is a stopped callback here, and only if there
+              // wasn't it would PostStop signal the previous behavior, now the stop behavior
+              // may be wrapped in any number of interceptors though, so doing that is tricky
+              // therefore currently both the previous behavior and the returned get the signal
+              // maybe that is ok, or maybe it is problematic, not sure
+              Behavior.interpretSignal(lastBehavior, ctx, PostStop)
+              Behavior.interpretSignal(s, ctx, PostStop)
+            }
```

Comment:
Since the behavior can be wrapped in any number of interceptors, it is hard to pull the OptionVal callback out of there, so instead I made it invoke both that and the callback.

This means such a behavior:
```
Behavior.receiveMessage { 
  case "stop" =>
    Behavior.stop(() => bö)
} receiveSignal {
  case (_, PostStop) =>
    bo
}
```

Will execute both `bo` and `bö`. This didn't break any tests but maybe there are scenarios where there is a problem.

Reply:
That doesn't look too bad to me, given that it is now a callback.

Reply:
Yeah, I have thought about it since pushing and I can't think of any scenario where it is very surprising. Let's go with this!

Reply:
I found a scenario where it is surprising now - with an interceptor on the previous behavior, the interceptor is kept to the stopping behavior and will get the PostStop on both. I haven't quite gotten my head around this issue - is it ok - should we make sure it doesn't happen etc.